tests/spec/context_spec.lua

```
@@ -0,0 +1,296 @@
+  it("should collect the entire buffer when no selection is made", function()
+    -- Mock buffer state
+    vim.bo[1] = { filetype = "lua" }
+    
+    local result = context.collect()
+    
+    assert.is_not_nil(result, "Context should not be nil")
+    assert.equals(result.language, "lua", "Language should be detected as lua")
+    assert.is_not_nil(result.content, "Content should be present")
+    assert.equals(result.file_path, "/test/file.lua", "File path should be correct")
+  end)
+  
```

Luacheck (1.2.0)

[warning] line contains only whitespace

(W611)